src/nvim/ops.c

```diff
@@ -5404,9 +5411,60 @@ static bool get_clipboard(int name, yankreg_T **target, bool quiet)
   return false;
 }
 
+
+/*
+ * Save and restore clip_unnamed before doing possibly many changes. This
+ * prevents accessing the clipboard very often which might slow down Vim
+ * considerably.
+*/
+
+/*
+ * Save clip_unnamed and reset it.
+ */
+void start_global_changes(void)
+{
+  if (++global_change_count > 1) {
+    return;
+  }
+  clip_unnamed_saved = clip_unnamed;
+  clipboard_needs_update = FALSE;
+
+  if (clip_did_set_selection) {
+    clip_unnamed = 0;
+    clip_did_set_selection = FALSE;
+  }
+}
+
+/*
+ * Restore clip_unnamed and set the selection when needed.
+ */
+void end_global_changes(void)
+{
+  if (--global_change_count > 0) {
+    // recursive
+    return;
+  }
+  if (!clip_did_set_selection) {
+    clip_did_set_selection = TRUE;
+    clip_unnamed = clip_unnamed_saved;
+    clip_unnamed_saved = FALSE;
+    if (clipboard_needs_update) {
+      if (clip_unnamed & CB_UNNAMED)
+      {
+	  set_clipboard('*', &y_regs[STAR_REGISTER]);
```

Comment:
During global changes `"*` and `"+` are not used (do not change). Setting the clipboard with the same content would be unnecessary. Shoudn't we use register `""` to update register `"*` and/or `"+`?